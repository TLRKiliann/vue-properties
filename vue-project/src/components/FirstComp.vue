<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({
  name: "FirstComp",
  props: ['name'],
  data() {
    return {
      className: 'className'
    }
  },
  methods: {},
})
</script>

<template>
  <div class="div--h2">
    <h2 v-bind:class="className">Name : {{ name }}</h2>
  </div>
</template>

<style scoped>
.className {
    color: lightgreen;
  }
</style>
