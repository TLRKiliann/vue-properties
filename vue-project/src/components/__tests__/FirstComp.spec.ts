import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import FirstComp from '../FirstComp.vue'

describe('HelloWorld', () => {
  it('snapshot UI testing', () => {
    const wrapper = mount(FirstComp,{});
    expect(wrapper.text()).toMatchSnapshot()
  })
  it('renders properly', () => {
    const wrapper = mount(FirstComp, { props: { name: 'Hello Vitest' } })
    expect(wrapper.text()).toContain('Hello Vitest')
  })
})
