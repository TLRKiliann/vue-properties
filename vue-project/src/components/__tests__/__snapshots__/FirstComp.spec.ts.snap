// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > snapshot UI testing 1`] = `"Name :"`;
